packages/keystatic/src/app/ItemPage.tsx

```diff
@@ -882,37 +882,53 @@ function ItemPageWrapper(props: ItemPageWrapperProps) {
 
   const yjsInfo = useYjsIfAvailable();
 
+  const isItemDataLoading = itemData.kind !== 'loaded';
+  const isItemNotFound = !isItemDataLoading && itemData.data === 'not-found';
   const mapData = useData(
-    useCallback(async () => {
+    useCallback(() => {
       if (!yjsInfo) return;
       if (yjsInfo === 'loading') return LOADING;
-      await yjsInfo.doc.whenSynced;
-      if (itemData.kind !== 'loaded') return LOADING;
-      if (itemData.data === 'not-found') return;
-      let doc = yjsInfo.data.get(key);
-      if (doc instanceof Y.Doc) {
-        const promise = doc.whenLoaded;
-        doc.load();
-        await promise;
-      } else {
-        doc = new Y.Doc();
-        yjsInfo.data.set(key, doc);
-      }
-      const data = doc.getMap('data');
-      if (!data.size) {
-        const {
-          data: { initialState },
-        } = itemData;
-        doc.transact(() => {
-          for (const [key, value] of Object.entries(collectionConfig.schema)) {
-            const val = getYjsValFromParsedValue(value, initialState[key]);
-            data.set(key, val);
-          }
-        });
-      }
-      return data;
-    }, [collectionConfig, itemData, key, yjsInfo])
+      if (isItemDataLoading) return LOADING;
+      if (isItemNotFound) return;
+      return (async () => {
+        await yjsInfo.doc.whenSynced;
+        let doc = yjsInfo.data.get(key);
+        if (doc instanceof Y.Doc) {
+          const promise = doc.whenLoaded;
+          doc.load();
+          await promise;
+        } else {
+          doc = new Y.Doc();
+          yjsInfo.data.set(key, doc);
+        }
+        const data = doc.getMap('data');
+
+        return data;
+      })();
+    }, [isItemDataLoading, isItemNotFound, key, yjsInfo])
   );
+  useMemo(() => {
```

Comment:
Interesting, I've never seen `useMemo` w/o a returned value. Is there a benefit over `useEffect` here?

Reply:
I want it to happen in render rather than in an effect